bigtable/client/rpc_backoff_policy.cc

```diff
@@ -46,12 +46,25 @@ void ExponentialBackoffPolicy::setup(grpc::ClientContext& /*unused*/) const {}
 std::chrono::milliseconds ExponentialBackoffPolicy::on_completion(
     grpc::Status const& status) {
   using namespace std::chrono;
-  // TODO(#57) - we also need to randomize the sleep period.
   auto delay = duration_cast<milliseconds>(current_delay_);
-  current_delay_ *= 2;
-  if (current_delay_ >= maximum_delay_) {
-    current_delay_ = maximum_delay_;
+  std::uniform_int_distribution<int> rng_distribution(0,
+                                                      current_delay_.count());
+  // Randomized sleep period because it is possible that after some time all
+  // client have same sleep period if we use only exponential backoff policy.
+  auto random_delay = microseconds(rng_distribution(generator_));
+  random_delay *= 2;
+  if (random_delay <= current_delay_) {
+    std::uniform_int_distribution<int> distribution(
+        random_delay.count(), 3 * (random_delay.count()) / 2);
+    random_delay = microseconds(distribution(generator_));
+    random_delay *= 2;
   }
+  if (current_delay_ > maximum_delay) {
+    std::uniform_int_distribution<int> distribution(
+        3 * maximum_delay.count() / 4, 5 * maximum_delay.count() / 4);
```

Comment:
That `maximum_delay` variable (without the trailing `_`) is the default maximum delay defined in the anonymous namespace,  I think you want `maximum_delay_` (with a trailing `_`) because you want the maximum as defined for the current object, not the default value.

Please take the time to rename the variable holding the default (`default_maximum_delay`) seems appropriate, because it is clearly a bad choice I made.

Reply:
Fixed.